moves.pl

```diff
@@ -17,15 +17,19 @@
 % Matches for the slots following the location of the moving marble, according to the
 % given direction.
 slots_sequence_by_direction(BoardState, BoardSize, SourceRow, SourceCol, Direction, [SlotType|SlotTypes]):-
-    between(0, BoardSize, SourceRow),
-    between(0, BoardSize, SourceCol), !,
+    LateralBorderRowIndex is BoardSize + 1,
```

Comment:
Can you make sure this change doesn't break the code that checks for legal moves?

Reply:
On the contrary, the earlier code was broken, this is a fix.
For example, when we check if a move is legal after using the old code, we could get:
`slots_seq...(..., ['W', 'W', 'B']` (since it does not include the outer border
and then
`legal_move(['W', 'W', 'B'])` is false (though we could push enemy to border